website/docs/r/custom_ip_prefix.html.markdown

```diff
@@ -0,0 +1,85 @@
+---
+subcategory: "Network"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_custom_ip_prefix"
+description: |-
+  Manages a Custom Ip Prefix.
+
+---
+
+# azurerm_custom_ip_prefix
+
+Manages a Custom Ip Prefix.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_custom_ip_prefix" "example" {
+  name                  = "example-CustomIpPrefix"
+  location              = azurerm_resource_group.example.location
+  resource_group_name   = azurerm_resource_group.example.name
+  cidr                  = "0.0.0.0/24"
+  action                = "Provision"
+  authorization_message = "00000000-0000-0000-0000-000000000000|0.0.0.0/24|20991212"
+  signed_message        = "singed message for WAN validation"
+  zones                 = ["1", "2", "3"]
+
+  tags = {
+    env = "test"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Custom Ip Prefix. Changing this forces a new resource to be created.
+
+* `location` - (Required) The location where the Custom Ip Prefix should exist. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group in which to create the Custom Ip Prefix. Changing this forces a new resource to be created.
+
+* `cidr` - (Required) The `cidr` of the Custom Ip Prefix. Changing this forces a new resource to be created.
+
+-> **NOTE:** Currently, the `cidr` only supports IPv4.
+
+* `zones` - (Required) A list of availability zones which the Custom Ip Prefix should be allocated. Possible values are `1`, `2`, `3`. Changing this forces a new resource to be created.
+
+* `authorization_message` - (Optional) The authorization message for WAN validation. Changing this forces a new resource to be created.
```

Comment:
could we be more explicit in the name as what this should be
```suggestion
* `wan_authorization_message` - (Optional) The authorization message for WAN validation. Changing this forces a new resource to be created.
```

and given it is composed of 3 individual parts could we break it out into a block?

Reply:
The `authorization_message` consists of subscription Id, cidr and roa expiration date. Now that the `cidr` property is  exposed, and the subs id must also be the subs id that created the custom ip prefix resource, maybe just exposing the `roa_expiration_date` property is enough, WDYT? Any feedback feel free let me know, thanks:).